core/src/main/java/org/elasticsearch/search/suggest/Suggest.java

```diff
@@ -177,7 +178,12 @@ public static Suggest fromXContent(XContentParser parser) throws IOException {
         ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.currentToken(), parser::getTokenLocation);
         List<Suggestion<? extends Entry<? extends Option>>> suggestions = new ArrayList<>();
         while ((parser.nextToken()) != XContentParser.Token.END_OBJECT) {
-            suggestions.add(Suggestion.fromXContent(parser));
+            ensureExpectedToken(XContentParser.Token.FIELD_NAME, parser.currentToken(), parser::getTokenLocation);
+            ensureExpectedToken(XContentParser.Token.START_ARRAY, parser.nextToken(), parser::getTokenLocation);
+            Suggestion<? extends Entry<? extends Option>> suggestion = Suggestion.fromXContent(parser);
+            if (suggestion != null) {
```

Comment:
I took a look, the suggest output currently looks like:
```
"suggest": {
		"suggestType#my-suggest-1": [{ ...entry...}, { ...entry...}],
		"suggestType2#my-suggest-2": [...],
		"suggestType3#my-suggest-3": [...],
		...
	}
```
I find it highly unlikely we are adding any other array field here that isn't going to be another suggestion, so I will add the error in this case.